### PR TITLE
Implementation of catalog feature

### PR DESCRIPTION
**Pull request by tlelv**
The catalog feature displays a catalog of the given dataset, which can be either timeseries data or location data. As such, it has been added to each respective file.

### CODE REVIEW

File: cwms/catalog/catalog.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        """
+
+    if dataset is None:
+        raise ValueError("Cannot retrieve a time series for a catalog without a dataset")
+
+    endpoint = f"catalog/{dataset}"
+
+    # creates the dataframe from the timeseries data
+    response = api.get(endpoint=endpoint, params=params, api_version=1)
+    return Data(response, selector="time-series-catalog")
```
Comment (Enovotny) [inline cwms/catalog/catalog.py:56]: is the selector of 'time-series-catalog' the same for the location catalog?  Please make sure to test both.  now that I see everything I think we remove the get_catalog function.  Move the get_locations_catalog and get_timeseries_catalog into the catalog.py and have them both just call the api.get...  That way you can have the selector correct for each individual function.  If you have questions on that let me know.

**Review by Enovotny — CHANGES_REQUESTED** [review]

File: cwms/catalog/catalog.py (hunk 1 above)
Comment (tlelv) [reply cwms/catalog/catalog.py:56]: I apologize for my oversight and understand the problem. An alternative solution I have thought of is creating a check that sees whether it is a timeseries, or locations call and changes the selector while keeping the rest of the code. As an example: 

# Code
```
selector = "catalog"
    if dataset == "TIMESERIES":
        selector = "time-series-" + selector
    elif dataset == "LOCATIONS":
        selector = "locations-"

    return Data(response, selector=selector)
```

This keeps each respective function in their respective files (get_locations_catalog being in the locations directory). I have no opinion whether they should be all in the catalog directory or in their respective so I can adjust to what you believe is best.

Comment (Enovotny) [reply cwms/catalog/catalog.py:56]: lets put the functions in the catalog and leave them as stand along. Adding the if statement would work but adds complexity for testing.  Keeping the functions as stand alone I think is best.

File: cwms/locations/physical_locations.py [hunk 2]
```diff
@@ -4,9 +4,36 @@
...
 
 import cwms.api as api
+import cwms.catalog.catalog as ct
 from cwms.types import Data
 
 
+def get_locations_catalog(page: Optional[str] = None,
+                          page_size: Optional[int] = 5000,
+                          unit_system: Optional[str] = None,
+                          office: Optional[str] = None,
```
Comment (Enovotny) [inline cwms/locations/physical_locations.py:14]: also lets have the cwms-python use office_id.  I know that CDA goes back and forth. but for consistency lets have the parameters for cwms-python all be office_id.

**Review by Enovotny — APPROVED** [review]

**Pull request merged**